### PR TITLE
feat: add glint-language-server support

### PR DESCRIPTION
**Pull request by muziejus**
I've tried to implement glint-language-server support. It's a little tricky, because the glint documentation advises _against_ installing glint globally, so I had to come up with a different way to find the binary.

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
> because the glint documentation advises against installing glint globally, 

this is a more proper thing to do anyway, because you can't know if you're in a monorepo and potentially have different language-server versions present.

File: lua/lspconfig/server_configurations/glint.lua [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+return {
+  default_config = {
+    cmd = cmd,
+    filetypes = {
+      'html.handlebars',
+      'handlebars',
+      'typescript',
+      'typescript.glimmer',
+      'javascript',
+      'javascript.glimmer',
```
Comment (NullVoxPopuli) [inline lua/lspconfig/server_configurations/glint.lua:19]: for other reviewers, javascript.glimmer and typescript.glimmer are the filetypes assigned to `*.gts` and `*.gjs` in treesitter.

File: lua/lspconfig/server_configurations/glint.lua [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+-- Glint should not be installed globally.
+local path_to_node_modules = util.find_node_modules_ancestor(vim.fn.getcwd())
+
+local cmd = { path_to_node_modules .. '/node_modules/.bin/' .. bin_name }
+
+return {
+  default_config = {
+    cmd = cmd,
+    filetypes = {
+      'html.handlebars',
```
Comment (NullVoxPopuli) [inline lua/lspconfig/server_configurations/glint.lua:14]: for other reviewers, html.handlebars is the filetype assigned to hbs files from vim-ember-hbs (a syntax-highlighting plugin)

File: lua/lspconfig/server_configurations/glint.lua [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  docs = {
+    description = [[
+  https://github.com/typed-ember/glint
+
+  https://typed-ember.gitbook.io/glint/
+
+  `glint-language-server` is installed when adding `@glint/core` to your project's devDependencies:
+
+  ```sh
+  yarn add -D @glint/core
```
Comment (NullVoxPopuli) [inline lua/lspconfig/server_configurations/glint.lua:32]: should we add npm and pnpm instructions here as well?

**Comment by muziejus** [comment]
Right, I'm trying to recreate this recommended functionality:

> It will open or create a .vim/coc-settings.json file. Add this config:
> 
> ```json
> { 
>   "tsserver.enable": false,
>   "languageserver": {
>     "glint": {
>       "command": "${workspaceFolder}/node_modules/.bin/glint-language-server",
>       "filetypes": ["html.handlebars","typescript"]
>     }
>   }
> }
> ```
>
> This will disable coc-tssever (if installed) and configure CoC to use Glint for .hbs and .ts files.

From [this blog post on glint](https://www.salsify.com/blog/engineering/introducing-glint-a-typed-template-solution-for-glimmerx-and-ember)

**Comment by NullVoxPopuli** [comment]
Testing this locally (with packer): 
```diff
-  -- use 'neovim/nvim-lspconfig'
+  use { 'muziejus/nvim-lspconfig', branch = "glint" }
```

**Comment by NullVoxPopuli** [comment]
Tested this over in: https://github.com/gitKrystan/super-rentals-ts-2022/pull/1

And it looks like it works:
image

**Comment by NullVoxPopuli** [comment]
Regular typescript stuff still works, too: 
image

**Review by justinmk — APPROVED** [review]

**Comment by muziejus** [comment]
@justinmk I removed the capital letter from the commit message.

**Pull request merged**